examples/cloudml-churn-prediction/preprocessor/preprocessor/preprocess.py

```diff
@@ -0,0 +1,325 @@
+    )
+    parser.add_argument(
+        '--log_level',
+        help='Set logging level',
+        default='INFO'
+    )
+    parser.add_argument(
+        '--bq_table',
+        help="""Source BigQuery table in [Project]:[Dataset]:[Table]
+            format.""",
+        default='bigquery-public-data.google_analytics_sample.ga_sessions_*'
+    )
+    parser.add_argument(
+        '--machine_type',
+        help="""Set machine type for Dataflow worker machines.""",
+        default='n1-highmem-4'
+    )
+    parser.add_argument(
+        '--cloud',
+        help="""Run preprocessing on the cloud. Default False.""",
+        action='store_true',
+        default=False,
+    )
+    parser.add_argument(
+        '--project_id',
+        help="""Google Cloud project ID""",
+        required=True,
+    )
+    known_args, _ = parser.parse_known_args(argv)
+
+    return known_args
+
+
+def get_pipeline_args(flags):
+    """Create Apache Beam pipeline arguments"""
+    options = {
+        'project': flags.project_id,
+        'staging_location': os.path.join(flags.output_dir, 'staging'),
+        'temp_location': os.path.join(flags.output_dir, 'temp2'),
+        'job_name': flags.job_name,
+        'save_main_session': True,
+        'setup_file': './setup.py'
+    }
+    return options
+
+
+def run():
+    """Run Apache Beam pipeline to generate TFRecords for Survival Analysis"""
+    flags = parse_arguments(sys.argv[1:])
+    pipeline_args = get_pipeline_args(flags)
+
+    options = PipelineOptions(flags=[], **pipeline_args)
+    options.view_as(WorkerOptions).machine_type = flags.machine_type
+
+    input_spec = features.RAW_FEATURE_SPEC
+    temp_dir = os.path.join(flags.output_dir, 'tmp')
+
+    runner = 'DataflowRunner' if flags.cloud else 'DirectRunner'
+
+    with beam.Pipeline(runner, options=options) as p:
+        with tft_beam.Context(temp_dir=temp_dir):
+            raw_data = (p 
+                        | 'QueryTable' >> beam.io.Read(beam.io.BigQuerySource(
+                            query=query.get_query(flags.bq_table),
+                            project=flags.project_id,
+                            use_standard_sql=True))
+                        #omit 'Generate Data' step if working with real data
+                        | 'Generate Data' >> 
+                            beam.Map(lambda row: _generateFakeData(row))
+                        | 'Extract duration and Label' >> 
+                            beam.Map(lambda row: _mapToClass(row))
+                        | 'Generate label array' >> 
+                            beam.Map(lambda row: 
+                                _combineCensorshipDuration(row))
+                       )
+            raw_train, raw_eval, raw_test = (
+                raw_data | 'RandomlySplitData' >> randomly_split(
+                    train_size=.7,
+                    validation_size=.15,
+                    test_size=.15))
+            raw_metadata = features.get_raw_dataset_metadata()
```

Comment:
Optional: I think the tft logic could be pulled into a helper function so `run()` is more readable